packages/apputils/src/clientsession.ts

```diff
@@ -866,6 +872,43 @@ namespace Private {
    */
   export
   function selectKernel(session: ClientSession): Promise<Kernel.IModel> {
+    let select = Dialog.okButton({ label: 'SELECT' });
+    return showDialog({
+      title: 'Select Kernel',
+      body: new KernelSelector(session),
+      buttons: [Dialog.cancelButton(), select]
+    }).then(result => {
+      if (!result.accept) {
+        return void 0;
```

Comment:
the return type should be `Promise<Kernel.IModel | null>` and then you can just `return result.value`, or this `undefined` needs to be replaced with an actual model.